packages/flutter_tools/lib/src/commands/build_aot.dart

```diff
@@ -0,0 +1,129 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:path/path.dart' as path;
+
+import '../base/process.dart';
+import '../build_configuration.dart';
+import '../globals.dart';
+import '../runner/flutter_command.dart';
+import '../toolchain.dart';
+import 'run.dart';
+
+const String _kDefaultAotOutputDir = 'build/aot';
+
+class BuildAotCommand extends FlutterCommand {
+  BuildAotCommand() {
+    usesTargetOption();
+    addBuildModeFlags();
+    usesPubOption();
+    argParser.addOption('output-dir', defaultsTo: _kDefaultAotOutputDir);
+  }
+
+  @override
+  final String name = 'aot';
+
+  @override
+  final String description = "Build an ahead-of-time compiled snapshot of your app's Dart code.";
+
+  @override
+  Future<int> runInProject() async {
+    String outputPath = buildAotSnapshot(
+      findMainDartFile(argResults['target']),
+      outputPath: argResults['output-dir']
+    );
+    if (outputPath == null)
+      return 1;
+
+    printStatus('Built $outputPath');
+    return 0;
+  }
+}
+
+String _getSdkExtensionPath(String packagesPath, String package) {
+  Directory packageDir = new Directory(path.join(packagesPath, package));
+  return path.join(path.dirname(packageDir.resolveSymbolicLinksSync()), 'sdk_ext');
+}
+
+String buildAotSnapshot(
+  String mainPath, {
+  String outputPath: _kDefaultAotOutputDir
+}) {
+  String engineSrc = ToolConfiguration.instance.engineSrcPath;
+  if (engineSrc == null) {
+    printError('AOT compilation requires --engine-src-path');
+    return null;
+  }
+
+  String engineOut = ToolConfiguration.instance.getEngineArtifactsDirectory(
+      TargetPlatform.android_arm, BuildMode.profile).path;
+  String genSnapshot = path.join(engineOut, 'clang_x86', 'gen_snapshot');
+
+  Directory outputDir = new Directory(outputPath);
+  outputDir.createSync(recursive: true);
+  String vmIsolateSnapshot = path.join(outputDir.path, 'snapshot_aot_vmisolate');
+  String isolateSnapshot = path.join(outputDir.path, 'snapshot_aot_isolate');
+  String instructionsBlob = path.join(outputDir.path, 'snapshot_aot_instr');
+  String rodataBlob = path.join(outputDir.path, 'snapshot_aot_rodata');
+
+  String bindingsSrc = path.join(engineSrc, 'sky', 'engine', 'bindings');
+  String vmEntryPoints = path.join(bindingsSrc, 'dart_vm_entry_points.txt');
+  String vmEntryPointsAndroid = path.join(bindingsSrc, 'dart_vm_entry_points_android.txt');
+
+  String packagesPath = path.absolute(Directory.current.path, 'packages');
```

Comment:
We're not using `packages` symlinks anymore. You'll have to locate packages by parsing the `.packages` file. We have code to do that in `lib/src/package_map.dart`.